### PR TITLE
COMPAT: NaT support tz_localize / tz_convert (#15830)

### LINKED ISSUE #15830: COMPAT: NaT should support tz_localize / tz_convert
These should simply return a ``NaT``.

makes ``NaT`` more Timestamp accessible.

these are added like in #15782 

further would be nice to audit all methods that are not implemented to see if they should be added as well (again most of these will likely return ``NaT`` or ``np.nan``)

### PR DESCRIPTION
**Pull request by atbd**
closes #15830 

* add tz_convert/tz_localize methods
* add tests

 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``

### CODE REVIEW

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -845,6 +845,12 @@ class NaTType(_NaT):
         # GH 10939
         return np.nan
 
+    def tz_localize(self):
+        return NPY_NAT
```
Comment (jreback) [inline pandas/_libs/tslib.pyx]: this is not necessary (that's the point of the ``_nat_methods`` code below

**Comment by jreback** [comment]
can you add a whatsnew note as well.

**Comment by atbd** [comment]
Thanks, I erased the unnecessary methods and add the wathsnew note

**Comment by jreback** [comment]
thanks @atbd nice PR!

**Pull request merged**